File: 0050-powx-n/0050-powx-n.py
class Solution:
    def findPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1
        if n < 0:
            return self.findPow(1/x, -1 * n)
        if n % 2 == 0:
            return self.findPow(x * x, n / 2)
        else:
            return x * self.findPow(x * x, (n - 1) / 2)
    def myPow(self, x: float, n: int) -> float:
        return self.findPow(x, n)

File: 0050-powx-n/0050-powx-n.java
class Solution {
    public double myPow(double x, int n) {
       return findPow(x, (long) n);
    }

    public double findPow(double x, long n) {
        if (n == 0) {
            return 1;
        } else if (n < 0) {
            return 1 / findPow(x, -1 * n);
        } else if (n % 2 == 0) {
            return findPow(x * x, n / 2);
        } else {
            return x * findPow(x * x, (n - 1) / 2);
        }
    }
}